packages/inventory/src/Stocktake/DetailView/reducer.ts

```diff
@@ -45,13 +49,56 @@ const StocktakeActionCreator = {
       payload: { newStatus },
     };
   },
+  sortBy: (column: Column<StocktakeItem>): StocktakeAction => ({
+    type: StocktakeActionType.SortBy,
+    payload: { column },
+  }),
 };
 
 export const getInitialState = (): StocktakeStateShape => ({
   draft: placeholderStocktake,
   sortBy: { key: 'itemName', isDesc: false, direction: 'asc' },
 });
 
+const toLookup = <T, K extends keyof T & string>(
+  things: T[],
+  key: K
+): Record<string, T[]> => {
+  const lookup: Record<string, T[]> = {} as Record<string, T[]>;
+  things.forEach(thing => {
+    const value = String(thing[key]);
+    if (!lookup[value]) {
+      lookup[value] = [];
+    }
+    lookup[value].push(thing);
+  });
+  return lookup;
+};
+
+const createStocktakeItem = (
+  id: string,
+  lines: StocktakeLine[]
+): StocktakeItem => {
+  return {
+    id,
+    lines,
+    itemName: () => ifTheSameElseDefault(lines, 'itemName', ''),
+    itemCode: () => ifTheSameElseDefault(lines, 'itemCode', ''),
+    batch: () => ifTheSameElseDefault(lines, 'batch', '[multiple]'),
+    expiryDate: () => ifTheSameElseDefault(lines, 'expiryDate', '[multiple]'),
+    countedNumPacks: () =>
+      String(ifTheSameElseDefault(lines, 'countedNumPacks', '[multiple]')),
+    snapshotNumPacks: () =>
+      String(ifTheSameElseDefault(lines, 'snapshotNumPacks', '[multiple]')),
+  };
+};
```

Comment:
little bit of an experiment/iteration. I don't like trying to keep the summary item values in sync, so calculating each time seems better.. though I suspect that the closure will make this trickier than I would hope

Reply:
Ah, and the untranslated `[multiple]`  - I'm thinking of putting into the cell a check to see if the value is `[multiple]` and using a translated string  🤔